components/molecules/AlgoliaResults/Templates/SearchResults.js

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import {connectStateResults, InfiniteHits} from 'react-instantsearch-dom'
+import styles from '../AlgoliaResults.module.css'
+import Hit from './Hit'
+import NoResults from './NoResults'
+import PostType from '../Facets/PostType'
+import Authors from '../Facets/Authors'
+import CustomClearRefinements from '../Refinements/CustomClearRefinements'
+import Sort from '../Facets/Sort'
+
+/**
+ * Refinement config passed into Algolia facets.
+ */
+const refinements = {
+  limit: 3,
+  translations: {
+    showMore(expanded) {
+      return expanded ? 'Less' : 'More'
+    }
+  }
+}
+
+/**
+ * Component for rendering search results.
+ */
+const SearchResults = connectStateResults(({searchResults, indexName}) => {
+  return (
+    <>
+      {searchResults && searchResults.nbHits ? (
+        <>
+          <div className={styles.resultsHeader}>
+            <div>
+              <h1>Search Results</h1>
+              <p className={styles.total}>
+                <span>{searchResults.nbHits} Results</span> for{' '}
+                {searchResults.query}
+              </p>
+            </div>
+            <Sort index={indexName} />
+          </div>
+          <div className={styles.results}>
+            <aside className={styles.sidebar}>
+              <Authors refinements={refinements} />
+              <PostType refinements={refinements} />
+              <CustomClearRefinements clearsQuery={true} />
+            </aside>
+            <div className={styles.content}>
+              <InfiniteHits
+                className={styles.aisHits}
+                hitComponent={Hit}
+                translations={{
+                  loadMore: 'Load More'
+                }}
+              />
+            </div>
+          </div>
+        </>
+      ) : (
+        <></>
```

Comment:
@dcooney You've got an empty return here. Just making sure this is correct.

Reply:
@gregrickaby Unfortunately, It's correct - without the empty return the app returns 0 for empty search results. I'll take another stab at this though, I may be able to refactor this with some state.